File: FizzBuzz.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

void fizzbuzz(char* fb, int i) {
  sprintf(fb, "%s%s", i%3 ? "" : "Fizz", i%5 ? "" : "Buzz");
  if (fb[0] == '\0') sprintf(fb, "%d", i); 
}

int main() {
  char fb[9];
  for (int i=1; i<=100; i++) {
    fizzbuzz(fb, i);
    printf("%s\n", fb);
  }
  return 0;
}

File: FizzBuzz.java
public class FizzBuzz {
    public static String fizzBuzz(int num) {
        String fb = (num % 3 == 0 ? "Fizz" : "") + (num % 5 == 0 ? "Buzz" : "");
        if (fb.isEmpty()) return String.valueOf(num);
        else return fb;
    }

    public static void main(String[] args) {
        for (int number = 1; number <= 100; number++) {
            System.out.println(fizzBuzz(number));
        }
    }
}